locale/en/docs/guides/security/index.md

```diff
@@ -0,0 +1,439 @@
+new or existing files which are loaded.
+
+Node.js has an experimental[¹][experimental-features]
+[policy mechanism][] to declare the loaded resource as untrusted or trusted.
+However, this policy is not enabled by default.
+Be sure to pin dependency versions and run automatic checks for vulnerabilities
+using common workflows or npm scripts.
+Before installing a package make sure that this package is maintained and
+includes all the content you expected in.
+Be careful, the Github source code is not always the same as the published one,
+validate it in the _node_modules_.
+
+#### Supply chain attacks
+
+A supply chain attack on a Node.js application happens when one of its
+dependencies (either direct or transitive) is compromised.
+This can happen either due to the application being too lax on the specification
+of the dependencies (allowing for unwanted updates) and/or common typos in the
+specification (vulnerable to [typosquatting][]).
+
+An attacker that takes control of an upstream package can publish a new version
+with malicious code in it. If a Node.js application depends on that package
+without being strict on which version is safe to use, the package can be
+automatically updated to the latest malicious version, compromising the application.
+
+Dependencies specified in the `package.json` file can have an exact version number
+or a range. However, when pinning a dependency to an exact version, its
+transitive dependencies are not themselves pinned.
+This still leaves the application vulnerable to unwanted/unexpected updates.
+
+Possible vector attacks:
+
+* Typosquatting attacks
+* Lockfile poisoning
+* Compromised maintainers
+* Malicious Packages
+* Dependency Confusions
+
+**Mitigations**
+
+* Prevent npm from executing arbitrary scripts with `--ignore-scripts`
+  * Additionally, you can disable it globally with `npm config set ignore-scripts true`
+* Pin dependency versions to a specific immutable version,
+  not a version that is a range or from a mutable source.
+* Use lockfiles, which pin every dependency (direct and transitive).
+  * Use [Mitigations for lockfile poisoning][].
+* Automate checks for new vulnerabilities using CI, with tools like [`npm-audit`][].
+  * Tools such as [`Socket`][] can be used to analyze packages with static analysis
+  to find risky behaviors such as network or filesystem access.
+* Use [`npm ci`][] instead of `npm install`.
+This enforces the lockfile so that inconsistencies between it and the
+_package.json_ file cause an error (instead of silently ignoring the lockfile
+in favor of _package.json_).
+* Carefully check the _package.json_ file for errors/typos in the names of the
+dependencies.
+
+### Memory Access Violation (CWE-284)
+
+Memory-based or heap-based attacks depends on a combination of memory management
+errors and an exploitable memory allocator.
+Like all runtimes, Node.js is vulnerable to these attacks if your projects runs
+on a shared machine.
```

Comment:
I'm not sure why the `shared machine` is relevant here? Do you mean because Node.js uses libc and the libc could be vulnerable?

Reply:
No, it's because when sharing resources with other processes, the memory can be violated.